hardware_interface/include/hardware_interface/hardware_component_interface.hpp

```diff
@@ -40,13 +40,27 @@
 #include "rclcpp/logging.hpp"
 #include "rclcpp/node_interfaces/node_clock_interface.hpp"
 #include "rclcpp/time.hpp"
+#include "rclcpp/version.h"
 #include "rclcpp_lifecycle/node_interfaces/lifecycle_node_interface.hpp"
 #include "rclcpp_lifecycle/state.hpp"
 #include "realtime_tools/async_function_handler.hpp"
 
 namespace hardware_interface
 {
 
+static inline rclcpp::NodeOptions get_hardware_component_node_options()
```

Comment:
Maybe instead of having it as a static method. Have you thought of adding a virtual method like controllers?

Reply:
I'm not sure it buys you anything if it is a virtual method. I followed the controller manager pattern where it is a free function.

Reply:
I'm talking about the pattern similar to this one

https://github.com/ros-controls/ros2_control/blob/783558f1f308fe0e1cbe64e3a5a6b7d92bd2d60a/controller_interface/include/controller_interface/controller_interface_base.hpp#L215-L226

We can add this in the future if needed. I don't mind :)